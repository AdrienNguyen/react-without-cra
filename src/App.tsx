import React from 'react'
import Button from '@/components/button'
import Github from '@/static/icons/github.svg'
import Banner from '@/static/images/banner.png'
import '@/styles/App.scss'

const App: React.FC = () => {
    const onFocus = () => {}

    const onBlur = () => {}

    return (
        <React.Fragment>
            <Button danger>My Button</Button>
            <h1>This is the first time</h1>
            <h2>This is second time</h2>
            <img className="banner" src={Banner} />
            <Github />
            <input
                className="phuongna"
                type="text"
                id="input-1"
                onFocus={onFocus}
                onBlur={onBlur}
            />
        </React.Fragment>
    )
}

export default App
