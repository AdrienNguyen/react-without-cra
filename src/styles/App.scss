@import 'variables';
@import 'animations';
@import 'base';
@import 'functions';
@import 'grid';
@import 'mixins';
@import 'typography';
@import 'utilities';

@import '../common-components/**/*.scss';
@import '../layouts/**/*.scss';
@import '../modules/**/**/*.scss';
