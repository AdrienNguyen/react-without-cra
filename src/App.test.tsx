import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import App from '@/App'

const renderer = createRenderer()

it('renders <App /> without crashing', () => {
    renderer.render(<App />)
    const renderOutput = renderer.getRenderOutput()
    expect(renderOutput).toMatchSnapshot()
})
