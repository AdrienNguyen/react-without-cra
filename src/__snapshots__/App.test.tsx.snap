// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App /> without crashing 1`] = `
<React.Fragment>
  <Button
    danger={true}
  >
    My Button
  </Button>
  <h1>
    This is the first time
  </h1>
  <h2>
    This is second time
  </h2>
  <img
    className="banner"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <SvgrURL />
  <input
    className="phuongna"
    id="input-1"
    onBlur={[Function]}
    onFocus={[Function]}
    type="text"
  />
</React.Fragment>
`;
